owl/SOMA-IO.owl

```diff
@@ -83,38 +554,454 @@
     
 
 
-    <!-- http://www.ease-crc.org/ont/SOMA.owl#SoftwareImplementation -->
+    <!-- http://www.ease-crc.org/ont/SOMA.owl#InterfaceComponentRole -->
 
-    <owl:Class rdf:about="http://www.ease-crc.org/ont/SOMA.owl#SoftwareImplementation">
-        <rdfs:subClassOf rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#DigitalObject"/>
-        <rdfs:comment>An information realization in form of a executable computer program.</rdfs:comment>
-        <rdfs:label>Software implementation</rdfs:label>
+    <owl:Class rdf:about="http://www.ease-crc.org/ont/SOMA.owl#InterfaceComponentRole">
+        <owl:equivalentClass>
+            <owl:Class>
+                <owl:intersectionOf rdf:parseType="Collection">
+                    <rdf:Description rdf:about="http://www.ease-crc.org/ont/SOMA.owl#SoftwareRole"/>
+                    <owl:Restriction>
+                        <owl:onProperty rdf:resource="http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#isDefinedIn"/>
+                        <owl:allValuesFrom rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#InterfaceSpecification"/>
+                    </owl:Restriction>
+                </owl:intersectionOf>
+            </owl:Class>
+        </owl:equivalentClass>
+        <rdfs:subClassOf rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#SoftwareRole"/>
+        <rdfs:label xml:lang="en">Interface component role</rdfs:label>
     </owl:Class>
     
 
 
-    <!-- http://www.ease-crc.org/ont/SOMA.owl#SoftwarePackage -->
+    <!-- http://www.ease-crc.org/ont/SOMA.owl#InterfaceSpecification -->
 
-    <owl:Class rdf:about="http://www.ease-crc.org/ont/SOMA.owl#SoftwarePackage">
-        <rdfs:subClassOf rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#DigitalObject"/>
-        <rdfs:comment>A structured archive of software that also often contains data needed to run the software, and scripts to build and configure the software.</rdfs:comment>
-        <rdfs:label>Software package</rdfs:label>
+    <owl:Class rdf:about="http://www.ease-crc.org/ont/SOMA.owl#InterfaceSpecification">
+        <rdfs:subClassOf rdf:resource="http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#Norm"/>
```

Comment:
Norm or Design?

Reply:
My intuition would be Design, because a "social norm" sounds like something that organically grew from within the society. I moved it. The original reason for Norm was that Robert liked that one better.